src/utils/content-type.ts

```diff
@@ -33,3 +33,16 @@ export enum ContentType {
     lcppdf = "application/pdf+lcp",
     pdf = "application/pdf",
 }
+
+export const parseContentType = (RawContentType: string): ContentType | undefined => {
```

Comment:
In Thorium, handling / comparison of "content type" values is very inconsistent. There is now this new function `parseContentType()`, but elsewhere in the codebase there is `===`, `array.includes()`, `string.startsWith()`.

I did a quick search for `ContentType.xxx` in the TypeScript source tree to find all usages:

In addition to this PR's `dispatchOpdsLink()` function, there are the `findLink()` and `filterTypeLink()` functions, there is also the code inside `opdsRequest()` (`isPub` `isAuth` and `isFeed`), and in `OpdsService` there is `contentTypeisXml()` + `contentTypeisOpds()`, and the comparison code in `returnPublication()` (`renewLink.Type`), and `importFromLinkService()` with `isLcpFile`, `isEpubFile`, `isAudioBookPacked`, `isAudioBookPackedLcp`, `isHtml`, `isDivinaPacked`, `isPdf`, `isJson`.

So, I do not mind merging this PR, but isn't this an opportunity to harmonize the source code, to remove the inconsistencies?

What do you think?

Reply:
Maybe see #1252 : it's extract from it

Reply:
I see. Let's merge this PR then, and the WIP PR will address inconsistencies.

Reply:
👍